.gitignore

```diff
@@ -1,4 +1,222 @@
-build/
-node_modules/
+### Typescript ###
+dist/
```

Comment:
This is cool, but can you clear up the rest. I believe a .gitignore should only care about artifacts from project itself. IDE config files, OS files etc should be handled by the users own global .gitignore file.